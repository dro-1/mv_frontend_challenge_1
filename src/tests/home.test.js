import React from 'react'
import { shallow } from 'enzyme'
import { HomePage } from './../pages/home.page'

describe('Home Page', () => {
  it('correctly renders Home Page', () => {
    expect(shallow(<HomePage balance={5000} />)).toMatchSnapshot()
  })

  it('does NOT call the deposit function if amount is less than 0', () => {
    const deposit = jest.fn()

    const wrapper = shallow(<HomePage deposit={deposit} />)

    wrapper.find('#amount').simulate('change', { target: { value: '-2545' } })

    wrapper.find('#depositBtn').simulate('click', {
      preventDefault: () => null,
    })

    expect(deposit).toBeCalledTimes(0)
    expect(wrapper.containsMatchingElement(<p> Invalid Amount </p>)).toEqual(
      true,
    )
  })

  it('DOES call the deposit function with the right amount', () => {
    const deposit = jest.fn()

    const wrapper = shallow(<HomePage deposit={deposit} />)

    wrapper.find('#amount').simulate('change', { target: { value: '2545' } })

    wrapper.find('#depositBtn').simulate('click', {
      preventDefault: () => null,
    })

    expect(deposit).toBeCalledWith(2545)
    expect(
      wrapper.containsMatchingElement(<p> You have deposited $2545</p>),
    ).toEqual(true)
  })

  it('does NOT call the withdraw function if amount is less than 0', () => {
    const withdraw = jest.fn()

    const wrapper = shallow(<HomePage withdraw={withdraw} />)

    wrapper.find('#amount').simulate('change', { target: { value: '-2545' } })

    wrapper.find('#withdrawBtn').simulate('click', {
      preventDefault: () => null,
    })

    expect(withdraw).toBeCalledTimes(0)
    expect(wrapper.containsMatchingElement(<p> Invalid Amount </p>)).toEqual(
      true,
    )
  })

  it('does NOT call the withdraw function if withdrawal leads to a negative balance', () => {
    const withdraw = jest.fn()

    const wrapper = shallow(<HomePage balance={5000} withdraw={withdraw} />)

    wrapper.find('#amount').simulate('change', { target: { value: '7534' } })

    wrapper.find('#withdrawBtn').simulate('click', {
      preventDefault: () => null,
    })

    expect(withdraw).toBeCalledTimes(0)
    expect(
      wrapper.containsMatchingElement(<p> Insufficient balance </p>),
    ).toEqual(true)
  })

  it('DOES call the withdraw function if the checks passed', () => {
    const withdraw = jest.fn()

    const wrapper = shallow(<HomePage balance={5000} withdraw={withdraw} />)

    wrapper.find('#amount').simulate('change', { target: { value: '2500' } })

    wrapper.find('#withdrawBtn').simulate('click', {
      preventDefault: () => null,
    })

    expect(withdraw).toBeCalledWith(2500)
    expect(
      wrapper.containsMatchingElement(<p> You have withdrawn $2500</p>),
    ).toEqual(true)
  })
})
