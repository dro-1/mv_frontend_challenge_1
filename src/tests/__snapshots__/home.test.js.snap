// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page correctly renders Home Page 1`] = `
<styled.div>
  <Header />
  <styled.main>
    <p>
      <span>
        Your account balance is: 
      </span>
      <span>
        $
        5000
      </span>
    </p>
    <input
      id="amount"
      onChange={[Function]}
      placeholder="Amount"
      type="number"
      value="0"
    />
    <div
      className="buttons"
    >
      <button
        id="withdrawBtn"
        onClick={[Function]}
      >
        Withdraw
      </button>
      <button
        id="depositBtn"
        onClick={[Function]}
      >
        Deposit
      </button>
    </div>
  </styled.main>
</styled.div>
`;
